python/ray/train/mosaic/mosaic_predictor.py

```diff
@@ -0,0 +1,60 @@
+from ray.train.torch import TorchPredictor
+from ray.train.batch_predictor import BatchPredictor
+from ray.air.checkpoint import Checkpoint
+from pathlib import Path
+import torch
+
+
+class MosaicPredictor(BatchPredictor):
```

Comment:
Why are we wrapping around BatchPredictor here? This is not a pattern we have anywhere else. Can we subclass TorchPredictor instead?

Reply:
I wrapped around BatchPredictor considering the usability. In most cases prediction would be made via BatchPredictor, and even when we wrap around TorchPredictor, it would not provide any more useful information. The main problem with getting a predictor for Mosaic Composer is that the checkpoint has to be reformatted, and so by directly wrapping around BatchPredictor, we can more efficiently reformat the checkpoint and instantiate BatchPredictor right away

Reply:
I am not sure that's the pattern we want (every other framework passes their predictor to batch predictor), but happy to discuss this offline

Reply:
MosaicPredictor has been updated